client/landing/stepper/declarative-flow/newsletter.ts

```diff
@@ -93,14 +106,14 @@ const newsletter: Flow = {
 			recordSubmitStep( providedDependencies, '', flowName, _currentStep );
 			const logInUrl = getStartUrl();
 
+			if ( ! userIsLoggedIn ) {
+				clearSignupDestinationCookie();
+				return window.location.assign( logInUrl );
+			}
+
 			switch ( _currentStep ) {
 				case 'intro':
-					clearSignupDestinationCookie();
```

Comment:
In the old version, clearSignupDestinationCookie() was invoked whether a user was logged in or not. In the new version, it is only invoked if the user is NOT logged in. Not sure if there's any concern about that difference or not?

Reply:
I went ahead and made this change.